src/dvc_objects/transfer.py

```diff
@@ -0,0 +1,39 @@
+from typing import Set
+
+from .db import ObjectDB
+
+
+def transfer(
+    src: "ObjectDB", dest: "ObjectDB", oids: Set["str"], jobs: int = None
+) -> Set["str"]:
+    src_exists = set(src.oids_exist(oids, jobs=jobs))
+    src_missing = oids - src_exists
+
+    dest_exists = set(dest.oids_exist(oids, jobs=jobs))
+    dest_missing = oids - dest_exists
+
+    fails = dest_missing & src_missing
+    new = src_exists - dest_exists
+
+    for oid in new:
+        path = src.oid_to_path(oid)
+        try:
+            dest.add(path, src.fs, oid)
+        except:  # pylint: disable=bare-except # noqa: B001, E722
+            fails.add(oid)
+
+    if fails:
+        raise Exception("failed to transfer", fails)
+    return new
+
+
+if __name__ == "__main__":
+    from dvc_objects.fs.implementations.memory import MemoryFileSystem
+
+    fs = MemoryFileSystem()
+    odb1 = ObjectDB(fs, "odb1")
+    odb2 = ObjectDB(fs, "odb2")
+    fs.pipe_file("/odb1/12/34", b"1234")
+
+    transfer(odb1, odb2, {"1234", "5678"})
+    assert fs.cat_file("/odb2/12/34") == b"1234"
```

Comment:
Can we just put it into a test? I don't quite understand these code pieces, especially in such a simple project.

Reply:
yeah, I'll remove this later. I'm just using this while working.

Reply:
Ah, my apologies.